### PR TITLE
Update to ember 4.12

### PR DESCRIPTION
**Pull request by aliokan**
Related to OP-2990

Main changes:
- Update ember and ember-data to v4.12.2
- Bump to @appuniversum/ember-appuniversum 2.18.0
- [Add required relationship options](https://github.com/lblod/frontend-organization-portal/pull/579/commits/221a15cdcb06dac68e6befc1fe845f3b897791bf)
- Use native array methods on Ember "Array-like"  https://github.com/lblod/frontend-organization-portal/pull/579/commits/9d855dda45be67a01d773b977cf39a3024124172 https://github.com/lblod/frontend-organization-portal/pull/579/commits/2b16eb7eeee9e3bc0056dafb003fe6496885e652
- Bump to node v18 
- Add SCSS linter

TODO
- [x] Revert https://github.com/lblod/frontend-organization-portal/pull/579/commits/c66ecfaa2f1d0687cc6864d3d63a8862e508fbf1 before merge

### CODE REVIEW

**Review by mirdono — COMMENTED** [review]
Looks good overall. Some minor comments/questions but nothing blocking imo

File: app/templates/administrative-units/administrative-unit/change-events/details/edit.hbs [hunk 1]
```diff
@@ -162,7 +162,7 @@
                 <Item>
                   <:label>Resulterende organisatie</:label>
                   <:content>
-                    {{@model.changeEvent.resultingOrganizations.firstObject.name}}
+                    {{get @model.changeEvent.resultingOrganizations "0.name"}}
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/templates/administrative-units/administrative-unit/change-events/details/edit.hbs:165]: It's a bit ugly, but it works ^^

File: app/components/administrative-unit-select-by-identifier.js [hunk 2]
```diff
@@ -50,7 +50,7 @@ export default class AdministrativeUnitSelectByIdentifierComponent extends Compo
     });
 
     if (result) {
-      return [...[searchParams], ...new Set(result.toArray())];
+      return [...[searchParams], ...new Set(result.slice())];
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/components/administrative-unit-select-by-identifier.js:53]: It's due to a deprecation added in 4.7: https://deprecations.emberjs.com/ember-data/v4.x#toc_ember-data-deprecate-array-like 
It was not mandatory to do the migration now, but it generated hundreds of warnings in the console.

File: app/routes/administrative-units/administrative-unit/related-organizations/edit.js [hunk 3]
```diff
@@ -18,18 +18,22 @@ export default class AdministrativeUnitsAdministrativeUnitRelatedOrganizationsEd
...
+        reload: true,
+        include: 'organization-status,was-founded-by-organizations',
+      }
+    );
 
-    const hasParticipants = (
-      await this.loadHasParticipantsTask.perform(administrativeUnit.id, params)
-    ).toArray();
+    const subOrganizations = await administrativeUnit.subOrganizations;
+    const hasParticipants = await administrativeUnit.hasParticipants;
```
[the remark this answers is not in the document]
Comment (aliokan) [reply app/routes/administrative-units/administrative-unit/related-organizations/edit.js:36]: Using `await` is an easy way to access the value of the relationship. Their is some explanations here https://rfcs.emberjs.com/id/0745-ember-data-deprecate-methods-on-promise-many-array

`administrativeUnit.subOrganizations` returns all the time a `PromiseManyArray`, whether the data is loaded or not. 

There is 2 scenarios: 
- the relationship is loaded, the promise is resolve instantaneously and return the value; 
- the relationship is not loaded, ember-data will load the data, the promise will resolve once the data is available. 

In some case, we can decide that we need all the time the relationship loaded with the model and declare it `sync` in the model, then the `await` is not needed, but we need to warranties that the relationship is loaded all the time, using `include` can be an option. Ember-data do not handle it by itself. 

One more thing, adding a lot of relationships in `include` reduce the number of calls to the server, but will increases the complexity on the server side.

File: app/components/administrative-unit-select-by-identifier.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (mirdono) [reply app/components/administrative-unit-select-by-identifier.js:53]: Cool, thanks for the links. Good to see things are being simplified upstream.

**Review by mirdono — APPROVED** [review]
Looks good to me.

**Pull request merged**